### PR TITLE
Add Apache Flink 1.12.2 release

### PR DESCRIPTION
**Pull request by rkhachatryan**
The PR adds 1.12.2 release by restoring a commit erroneously added the `asf-site` branch.

The site has not been rebuilt (changes from `bash docker-build.sh` are not included in the commit).

cc: @curcur @rmetzger @zentol

### CODE REVIEW

**Review by curcur — APPROVED** [review]

File: _posts/2021-02-16-release-1.12.2.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+- yuan:
+  name: "Yuan Mei"
+- roman:
+  name: "Roman Khachatryan"
+
+---
+
+The Apache Flink community released the next bugfix version of the Apache Flink 1.12 series.
+
+This release includes 69 fixes and minor improvements for Flink 1.12.1. The list below includes a detailed list of all fixes and improvements.
```
Comment (xintongsong) [inline _posts/2021-02-16-release-1.12.2.md]: JIRA shows 68 issues resolved in 1.12.2.
https://issues.apache.org/jira/projects/FLINK/versions/12349502

**Review by xintongsong — APPROVED** [review]
LGTM. Please double check the number and list of issues before merging.

File: _posts/2021-02-16-release-1.12.2.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-21048'>FLINK-21048</a>] -         Refactor documentation related to switch memory allocator 
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-21123'>FLINK-21123</a>] -         Upgrade Beanutils 1.9.x to 1.9.4
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-21164'>FLINK-21164</a>] -         Jar handlers don&#39;t cleanup temporarily extracted jars
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-21210'>FLINK-21210</a>] -         ApplicationClusterEntryPoints should explicitly close PackagedProgram
+</li>
+<li>[<a href='https://issues.apache.org/jira/browse/FLINK-21381'>FLINK-21381</a>] -         Kubernetes HA documentation does not state required service account and role
```
Comment (carp84) [inline _posts/2021-02-16-release-1.12.2.md]: This one (FLINK-21381) is committed to release-1.12 branch after RC1 is produced, and the current fix version of it is also 1.12.3, we should remove this one if the vote of RC1 passed.

File: _config.yml [hunk 3]
```diff
@@ -206,6 +206,10 @@ component_releases:
 release_archive:
     flink:
       - version_short: "1.12"
+        version_long: 1.12.2
+        release_date: 2021-02-16
```
Comment (carp84) [inline _config.yml]: Please remember to update the release date accordingly.

**Review by carp84 — APPROVED** [review]
Overall LGTM, please check the inline notes.

File: q/gradle-quickstart.sh [hunk 4]
```diff
@@ -41,7 +41,7 @@ function mkPackage() {
 defaultProjectName="quickstart"
 defaultOrganization="org.myorg.quickstart"
 defaultVersion="0.1-SNAPSHOT"
-defaultFlinkVersion="${1:-1.12.1}
+defaultFlinkVersion="${1:-1.12.2}
```
Comment (leewallen) [inline q/gradle-quickstart.sh]: This line is missing an ending quote and will cause the script to not work. I had entered a pull request for the 1.12.1 version to fix that.

Change this `defaultFlinkVersion="${1:-1.12.2}` to this `defaultFlinkVersion="${1:-1.12.2}"`

Comment (rkhachatryan) [reply q/gradle-quickstart.sh]: Thanks!

File: _config.yml (hunk 3 above)
Comment (rkhachatryan) [reply _config.yml]: I'll update the date before merging. Thanks!

File: _posts/2021-02-16-release-1.12.2.md (hunk 2 above)
Comment (rkhachatryan) [reply _posts/2021-02-16-release-1.12.2.md]: The issue is now included into RC#2.

**Comment by curcur** [comment]
I've reviewed the RC2 update-part, and doubled confirmed the change-list for 1.12.2

**Comment by rkhachatryan** [comment]
Rebased onto latest `apache/asf-site`

File: _posts/2021-02-16-release-1.12.2.md (hunk 1 above)
Comment (rkhachatryan) [reply _posts/2021-02-16-release-1.12.2.md]: Updated to 83 issues (RC2), thanks!

**Pull request merged**